cmd/inspect.go

```diff
@@ -161,55 +165,151 @@ func printContainerInspect(c *clab.CLab, containers []types.GenericContainer, fo
+		} else {
+			table.SetHeader(append([]string{"#"}, header[1:]...))
+		}
+		table.SetAutoFormatHeaders(false)
+		table.SetAutoWrapText(false)
+		// merge cells with lab name and topo file path
+		table.SetAutoMergeCellsByColumnIndex([]int{1, 2})
+		table.AppendBulk(tabData)
+		table.Render()
+
+		// do not print mysocket data if printMysocket is false or we don't have nodes populated
+		// nodes are not populated when `inspect --all` is used, since we don't read topology files
+		if !printMysocket || len(c.Nodes) == 0 {
+			return nil
+		}
+
+		// prepare data for table
+		var tabDataMySocketIo [][]string
+		for _, entry := range socketdata {
+			var portstrarr []string
+			for _, port := range entry.Ports {
+				portstrarr = append(portstrarr, strconv.Itoa(port))
+			}
+			tabDataMySocketIo = append(tabDataMySocketIo, []string{*entry.SocketId, *entry.DnsName, strings.Join(portstrarr, ", "), *entry.Type, strconv.FormatBool(entry.CloudAuth), *entry.Name})
+		}
+		tableMySocketIo := tablewriter.NewWriter(os.Stdout)
+		headerMySocketIo := []string{
+			"Socket ID",
+			"DNS Name",
+			"Ports",
+			"Type",
+			"Cloud Auth",
+			"Name",
+		}
+		// configure table output
+		tableMySocketIo.SetHeader(headerMySocketIo)
+		tableMySocketIo.SetAutoFormatHeaders(false)
+		tableMySocketIo.SetAutoWrapText(false)
+		tableMySocketIo.AppendBulk(tabDataMySocketIo)
+		fmt.Println("Published ports:")
+		tableMySocketIo.Render()
+
 		return nil
 	}
+	return nil
+}
 
-	runtime := c.GlobalRuntime()
+// getMySocketioData uses the mysocketio.http client to retrieve the socket data
+func getMySocketIoData(tokenfile string) ([]*types.MySocketIoEntry, error) {
+	result := []*types.MySocketIoEntry{}
 
-	stdout, stderr, err := runtime.Exec(context.Background(), mysocketCID, []string{"mysocketctl", "socket", "ls"})
+	client, err := mysocketio.NewClient(tokenfile)
 	if err != nil {
-		return fmt.Errorf("failed to execute cmd: %v", err)
+		return nil, err
+	}
 
+	sockets := []mysocketio.Socket{}
+	err = client.Request("GET", "connect", &sockets, nil)
+	if err != nil {
+		return nil, err
 	}
-	if len(stderr) > 0 {
-		log.Infof("errors during listing mysocketio sockets: %s", string(stderr))
+
+	for i := range sockets {
+		newentry := &types.MySocketIoEntry{
+			SocketId:  &sockets[i].SocketID,
+			DnsName:   &sockets[i].Dnsname,
+			Ports:     sockets[i].SocketTcpPorts,
+			Type:      &sockets[i].SocketType,
+			CloudAuth: sockets[i].CloudAuthEnabled,
+			Name:      &sockets[i].Name,
+		}
+		result = append(result, newentry)
 	}
+	return result, nil
+}
 
-	fmt.Println("Published ports:")
-	fmt.Println(string(stdout))
+// mySocketIoTokenFileFromBindMounts finds a node of kind mysocketio.
+// if that is found, the bindmounts are searched for ".mysocketio_token" and the path is being converted into an
+// absolute path and returned.
+func mySocketIoTokenFileFromBindMounts(_nodes map[string]nodes.Node, configPath string) (string, error) {
```

Comment:
Hi @steiler 
there is one little thing with this approach which I am ok to let slip

when you do `clab ins -a` this func will not work since we don't have `c.Nodes` populated as we don't read topology files.

One workaround for that is to use GenericContainer type where i added mounts information in this commit. We can rework this function to work on a slice of container info, rather than c.Nodes, and get the binds info from it. But then the problem is what to do if you have >1 mysocketio node, with different tokens...

So for the time being I just made a shunt so that mysocketdata is not fetched if we use `inspect --all`

Reply:
oh ok, I see.
What if we then simply print multiple tables indexed with either the containername or maybe better with the lab name, availabel via the container labels?!
In the json case, that would require a change to the output format as well ... so we eiter make the highlevel mysocketio entry a simple list, also carrying the labname as on item of the enclosed entries or a map with the labname as index... 
Thought?

Reply:
yes, I think adding lab name as a field for mysocket data is a valid approach. I would go with a list where lab name is a k:v pair inside the list

when `inspect --all` is used and the output is in the table format, then I think it is safe to have a single table that lists set up sockets with a field listing a lab for which these sockets were created.
we would just need to make sure we list through all containers which have `label[kind]=mysocketio`, get their respective source mount path and fetch the sockets.

Quite some work. Maybe better to do that in a separate PR?